### PR TITLE
Gradle include file for adding the Android dependency

### PR DESCRIPTION
**Pull request by cutiko**
The objective of this pull request is to have one source for content about adding the Android dependency. Before this, there were at least 3 files with repeated the content, inconsistent or used deprecated instructions.

By adding an include `_gradle.md`, maintenance can be done in one file only and avoids repetitive work in the rest of the documentation.

An include file name `_gradle.md` was created with the needed content. The file merge the instructions previously contained in:

 - `quickstart/native/android/_include/_auth0.md`
 - `quickstart/native/android/_include/_api_authz.md`

Some edition was made to the content to make it fit between both and modularly in any other file.

In the note used before there was a link to the Github documentation. The link was removed because the [documentation source file](https://raw.githubusercontent.com/auth0/Auth0.Android/master/README.md) has the version hardcoded. If the documentation referred is not updated it would be inconsistent with the Maven or JCenter latest version.

A minor note about this is that the block of code was using `xml` instead of `groovy` as code block language.

While checking if there was any other repetitive usage of the Android installation explained, the reference in the file `libraries/save-and-refresh-tokens.md` was found. It was using `compile` instead of `implementation`, the first is deprecated. Using the include `_gradle.md` the file allows keeping the reference updated in a single place.

The included usage was taken literally from the documentation, but in the files is different, so I went with my guts and stick to the guidelines.

I had problems understanding the [Wordy instructions for the editor](https://github.com/auth0/docs/blob/master/wordy-guide.md), so my apologies if anything failed on that step, however, it was followed and the editor's corrections were applied.

Update `yml` files are included.

If this approach please you, I found a similar problem for the manifest, I could take a look later. Please let me know if there is anything else I can help with.

---

https://docs-content-staging-pr-7900.herokuapp.com/docs/libraries/auth0-android
https://docs-content-staging-pr-7900.herokuapp.com/docs/libraries/auth0-android/save-and-refresh-tokens
https://docs-content-staging-pr-7900.herokuapp.com/docs/quickstart/native/android

### CODE REVIEW

File: articles/quickstart/native/android/_includes/_gradle.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  //Add the Auth0 Android SDK
+  implementation 'com.auth0.android:auth0:1.+'
+}
+```
+
+::: note
+If Android Studio lint the `+` sign, or if you want to use a fixed version, check for the latest in [Maven](http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22auth0%22%20g%3A%22com.auth0.android%22) or [JCenter](https://bintray.com/auth0/android/auth0).
+:::
+
+::: panel Synchronize Gradle
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_gradle.md]:
```suggestion
::: panel Sync Project with Gradle Files
```

File: articles/quickstart/native/android/_includes/_gradle.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Remember to synchronize using the Android Studio prompt or run `./gradlew clean assembleDebug` from the command line. For more information about Gradle usage, check [their official documentation](http://tools.android.com/tech-docs/new-build-system/user-guide).
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_gradle.md]: Let's change the command to `build`

```suggestion
Remember to synchronize using the Android Studio prompt or run `./gradlew clean build` from the command line. For more information about Gradle usage, check [their official documentation](http://tools.android.com/tech-docs/new-build-system/user-guide).
```

File: articles/quickstart/native/android/_includes/_gradle.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  //...
+}
+dependencies {
+  //Add the Auth0 Android SDK
+  implementation 'com.auth0.android:auth0:1.+'
+}
+```
+
+::: note
+If Android Studio lint the `+` sign, or if you want to use a fixed version, check for the latest in [Maven](http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22auth0%22%20g%3A%22com.auth0.android%22) or [JCenter](https://bintray.com/auth0/android/auth0).
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_gradle.md]: I think this should be `lints`

```suggestion
If Android Studio lints the `+` sign, or if you want to use a fixed version, check for the latest in [Maven](http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22auth0%22%20g%3A%22com.auth0.android%22) or [JCenter](https://bintray.com/auth0/android/auth0).
```

File: articles/quickstart/native/android/_includes/_gradle.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+## Add the Auth0 Android Dependency
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_gradle.md]: Remove this header, let's keep it on each of the files as we don't want to enforce a given header level (`##`). Some would not even use one.

File: articles/quickstart/native/android/_includes/_auth0.md [hunk 5]
```diff
@@ -1,26 +1,4 @@
-## Add the Auth0 Android Dependency
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_auth0.md]: Keep this header

File: articles/quickstart/native/android/_includes/_api_authz.md [hunk 6]
```diff
@@ -1,24 +1,4 @@
-## Configuring Your Application
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_api_authz.md]: Keep this header

File: articles/libraries/auth0-android/save-and-refresh-tokens.md [hunk 7]
```diff
@@ -17,11 +17,7 @@ When an authentication is performed with the `offline_access` <dfn data-key="sco
 
 [Auth0.Android](https://github.com/auth0/Auth0.Android) provides a utility class to streamline the process of storing and renewing credentials. You can access the `accessToken` or `idToken` properties from the [Credentials](https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/result/Credentials.java) instance. This is the preferred method to manage user credentials.
 
-First, add the library dependency to your build.gradle file:
-
-```gradle
-compile 'com.auth0.android:auth0:1.+'
-```
+<%= include('../../quickstart/native/android/_includes/_gradle.md') %>
```
Comment (lbalmaceda) [inline articles/libraries/auth0-android/save-and-refresh-tokens.md]: I'd remove this include and the previous dependency snippet as well. I'd just say something like 
>"Credential Managers are included as part of the Auth0.Android SDK. If this is not part of your dependencies yet, make sure to check {link to https://auth0.com/docs/libraries/auth0-android}".  

And BTW, I think we can also make use of this new include in the Installation section of that main SDK document. See https://auth0.com/docs/libraries/auth0-android#installation

**Review by lbalmaceda — CHANGES_REQUESTED** [review]
I think is a great change. I've left you some comments!

File: articles/libraries/auth0-android/save-and-refresh-tokens.md (hunk 7 above)
Comment (cutiko) [reply articles/libraries/auth0-android/save-and-refresh-tokens.md]: - Link added using markdown and following relative paths convention on contribution guide
- Previous content was replaced with the new `_gradle.md` without a title. The previous link to Gradle docs shouldn't be needed anymore and seems bigger than the intended scope.

**Comment by cutiko** [comment]
I agree with everything, thanks for pointing it out and for finding another source of improvement. My only doubt in this moment is:

Should we move the `_gradle.md` to `libraries/android/_includes/...` ?

**Comment by cutiko** [comment]
Greetings

Took the liberty of solving the merge conflict with my `yml` the other content should be included as well

I hope to help, let me know if anything else I can do.

**Comment by jeffreylees** [comment]
@lbalmaceda does this look good to you now?

**Review by lbalmaceda — DISMISSED** [review]

**Comment by lbalmaceda** [comment]
@cutiko Sorry I forgot about this one. It looks good now! Thanks for the changes.
@jeffreylees let me fix a file reference error and then this is ready to merge

**Comment by jeffreylees** [comment]
It builds now, and the pages load correctly - all good to merge @lbalmaceda ?

**Review by lbalmaceda — APPROVED** [review]

**Comment by jeffreylees** [comment]
Thanks @cutiko !!

**Pull request merged**